src/www/status_dhcpv6_leases.php

```diff
@@ -97,15 +97,49 @@ function parse_duid($duid_string) {
 $leasesfile = services_dhcpdv6_leasesfile();
 
 if ($_SERVER['REQUEST_METHOD'] === 'GET') {
-    $awk = "/usr/bin/awk";
-
-    /* this pattern sticks comments into a single array item */
-    $cleanpattern = "'{ gsub(\"^#.*\", \"\");} { gsub(\"^server-duid.*\", \"\");} { gsub(\";$\", \"\"); print;}'";
-    /* We then split the leases file by } */
-    $splitpattern = "'BEGIN { RS=\"}\";} {for (i=1; i<=NF; i++) printf \"%s \", \$i; printf \"}\\n\";}'";
-
-    /* stuff the leases file in a proper format into a array by line */
-    exec("/bin/cat {$leasesfile} | {$awk} {$cleanpattern} | {$awk} {$splitpattern} | /usr/bin/grep '^ia-.. '", $leases_content);
+    $leases_content = array();
+
+    $t_dhcpd6_leases = file_get_contents($leasesfile);
+    
+    // Set initial pointer position
+    $tpos = 0;
+    
+    // Lease of type 'ia-na'
+    while(1) {
+        $t_start_pointer = strpos($t_dhcpd6_leases,"ia-na",$tpos);
+        if($t_start_pointer == false) {
+            break;
+        }
+        // find end of lease entry, twice as all entries contain double brackets
+        $t_end_pointer = strpos($t_dhcpd6_leases,"}",$t_start_pointer);
+        $t_end_pointer = strpos($t_dhcpd6_leases,"}",$t_end_pointer+1);
+        // we have an entry. Now extract it.    
+        $t_entry = substr($t_dhcpd6_leases,$t_start_pointer, ($t_end_pointer+1) - $t_start_pointer);
+        // strip CR and semis
+        $t_entry = str_replace(";"," ", $t_entry);
+        $t_entry = str_replace("\n"," ", $t_entry);
+        // Put entry in array
+        array_push($leases_content,$t_entry);
+        // Update the pointer position for next search
+        $tpos = $t_end_pointer+1;
+    }
+    
+     $tpos = 0;
+    // Lease of type 'ia-pd'
+    while(1) {
+        $t_start_pointer = strpos($t_dhcpd6_leases,"ia-pd",$tpos);
+        if($t_start_pointer == false) {
+            break;
+        }
+        $t_end_pointer = strpos($t_dhcpd6_leases,"}",$t_start_pointer);
+        $t_end_pointer = strpos($t_dhcpd6_leases,"}",$t_end_pointer+1);
+        $t_entry = substr($t_dhcpd6_leases,$t_start_pointer, ($t_end_pointer+1) - $t_start_pointer);
+        $t_entry = str_replace(";"," ", $t_entry);
+        $t_entry = str_replace("\n"," ", $t_entry);
```

Comment:
I am not sure but you probably want to use this function:

https://secure.php.net/manual/de/function.rtrim.php

Reply:
If I use the fgets method yes.